File: exams/u3/sigcha/Exam 3/SortApplication/sort_app/model/bubble_sort.py
from typing import List
from sort_app.model.sorting_strategy import SortingStrategy

class BubbleSort(SortingStrategy):
    def sort(self, data: List[int]) -> List[int]:
        sorted_list = data[:]
        n = len(sorted_list)
        for i in range(n):
            for j in range(0, n-i-1):
                if sorted_list[j] > sorted_list[j+1]:
                    sorted_list[j], sorted_list[j+1] = sorted_list[j+1], sorted_list[j]
        return sorted_list

File: qz/u3/toscano/Evaluacion_Damian_Toscano/Evaluacion_Toscano/controller/BubbleSort.js
const SortingStrategy = require('./SortingStrategy');

class BubbleSort extends SortingStrategy {
    sort(data) {
        console.log("Bubble sort");
        let n = data.length;
        for (let i = 0; i < n - 1; i++) {
            for (let j = 0; j < n - i - 1; j++) {
                if (data[j] > data[j + 1]) {
                    [data[j], data[j + 1]] = [data[j + 1], data[j]];
                }
            }
        }
        return data;
    }
}

module.exports = BubbleSort;
